screens/modals/QuantityModal.js

```diff
@@ -0,0 +1,162 @@
+import { FontAwesome5 } from '@expo/vector-icons';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Modal, TouchableOpacity, View } from 'react-native';
+import Colors from '../../assets/Colors';
+import { Body, ButtonLabel, Title } from '../../components/BaseComponents';
+import LineItemCard from '../../components/LineItemCard';
+import ProductDisplayCard from '../../components/ProductDisplayCard';
+import {
+  ModalCenteredOpacityLayer,
+  ModalContentContainer,
+  ModalCopyContainer,
+  QuantityInput
+} from '../../styled/modal';
+import { ColumnContainer, RoundedButtonContainer } from '../../styled/shared';
+
+export default class QuantityModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalVisible: false,
+      product: null,
+      // Needs to be string type because it's TextInput
+      currentQuantity: '',
+      isLoading: true
+    };
+  }
+
+  componentDidMount() {
+    const { product } = this.props;
+    this.resetCurrentQuantity();
+    this.setState({
+      product,
+      isLoading: false
+    });
+  }
+
+  // Forces a re-render when new props are passed
+  componentWillReceiveProps(nextProps) {
+    const newQuantity = nextProps.product.quantity;
+    if (this.state.currentQuantity !== newQuantity) {
+      this.setState(prevState => ({
+        ...prevState,
+        product: nextProps.product,
+        currentQuantity: newQuantity === 0 ? '' : newQuantity.toString()
+      }));
+    }
+  }
+
+  resetCurrentQuantity = () => {
+    const { product } = this.props;
+    const quantityInt = product.quantity;
+    this.setState({ currentQuantity: quantityInt === 0 ? '' : quantityInt.toString() });
+  };
+
+  setModalVisible = visible => {
+    if (visible) {
+      this.resetCurrentQuantity();
+    }
+    this.setState({ modalVisible: visible });
+  };
+
+  handleShowModal = () => {
+    this.setModalVisible(!this.state.modalVisible);
```

Comment:
Again, any reason not to do `this.setModalVisible(true)` if this is always showing the modal (and not toggling it). I have NO idea how the UI works so please correct me!

Reply:
I might just suggest renaming to `toggleModal` as a more literal name

Reply:
This is a good point, and I was wondering myself... I actually already renamed `handleShowModal` to `toggleVisibility` in the next PR (LOL), will consider just replacing `setModalVisible` altogether. Prettyyyy sure the `handleShowModal()` was a) from boilerplate and b) used with conditions when I didn't know `disabled` was a prop for buttons (^: